components/jstraceable_derive/lib.rs

```diff
@@ -149,21 +149,14 @@ fn js_traceable_derive(s: synstructure::Structure) -> proc_macro2::TokenStream {
     let mut asserts = quote!();
     let match_body = s.each(|binding| {
         for attr in binding.ast().attrs.iter() {
-            match attr.parse_meta().unwrap() {
-                syn::Meta::Path(ref path) | syn::Meta::List(syn::MetaList { ref path, .. }) => {
-                    if path.is_ident("no_trace") {
-                        asserts.extend(assert_not_impl_traceable(&binding.ast().ty));
-                        return None;
-                    } else if path.is_ident("custom_trace") {
-                        return Some(quote!(<crate::dom::bindings::trace::CustomTraceable>::trace(#binding, tracer);));
-                    }
-                },
-                syn::Meta::NameValue(syn::MetaNameValue { ref path, .. }) => {
-                    // if reason provided we can skip JSTraceable check
-                    if path.is_ident("no_trace") {
-                        return None;
-                    }
-                },
+            if attr.path().is_ident("no_trace") {
+                // if reason is not provided we do JSTraceable check
+                if attr.meta.require_name_value().is_err() {
+                    asserts.extend(assert_not_impl_traceable(&binding.ast().ty));
+                }
```

Comment:
```suggestion
            if attr.path().is_ident("no_trace") {
                // Do a `JSTraceable` check unless a reason is provided with `no_check`.
                if attr.meta.require_name_value().is_err() {
                    asserts.extend(assert_not_impl_traceable(&binding.ast().ty));
                }
```

It's a bit odd to trigger an error here. Could this be:

```rust
if attr.meta != Meta::NameValue() {
 ...
}
```
?

I think avoiding the error makes things a bit clearer.

Reply:
matches! macro should do the job.